win32/CMakeLists.txt

```diff
@@ -236,36 +236,66 @@ if(USE_SE_SM)
-set(GLSLANG_SUFFIX "d")
-endif()
+add_library(glslangd STATIC IMPORTED GLOBAL)
+set_target_properties(glslangd PROPERTIES
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/glslangd.lib
+)
+add_library(HLSLd STATIC IMPORTED GLOBAL)
+set_target_properties(HLSLd PROPERTIES
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/HLSLd.lib
+)
+add_library(OGLCompilerd STATIC IMPORTED GLOBAL)
+set_target_properties(OGLCompilerd PROPERTIES
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/OGLCompilerd.lib
+)
+add_library(OSDependentd STATIC IMPORTED GLOBAL)
+set_target_properties(OSDependentd PROPERTIES
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/OSDependentd.lib
+)
+add_library(SPIRVd STATIC IMPORTED GLOBAL)
+set_target_properties(SPIRVd PROPERTIES
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/SPIRVd.lib
+)
+add_library(SPIRVRemapperd STATIC IMPORTED GLOBAL)
+set_target_properties(SPIRVRemapperd PROPERTIES
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/SPIRVRemapperd.lib
+)
+
 
 add_library(glslang STATIC IMPORTED GLOBAL)
 set_target_properties(glslang PROPERTIES
-  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/glslang${GLSLANG_SUFFIX}.lib
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/glslang.lib
 )
 add_library(HLSL STATIC IMPORTED GLOBAL)
 set_target_properties(HLSL PROPERTIES
-  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/HLSL${GLSLANG_SUFFIX}.lib
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/HLSL.lib
 )
 add_library(OGLCompiler STATIC IMPORTED GLOBAL)
 set_target_properties(OGLCompiler PROPERTIES
-  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/OGLCompiler${GLSLANG_SUFFIX}.lib
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/OGLCompiler.lib
 )
 add_library(OSDependent STATIC IMPORTED GLOBAL)
 set_target_properties(OSDependent PROPERTIES
-  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/OSDependent${GLSLANG_SUFFIX}.lib
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/OSDependent.lib
 )
 add_library(SPIRV STATIC IMPORTED GLOBAL)
 set_target_properties(SPIRV PROPERTIES
-  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/SPIRV${GLSLANG_SUFFIX}.lib
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/SPIRV.lib
 )
 add_library(SPIRVRemapper STATIC IMPORTED GLOBAL)
 set_target_properties(SPIRVRemapper PROPERTIES
-  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/SPIRVRemapper${GLSLANG_SUFFIX}.lib
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/libs/glslang/SPIRVRemapper.lib
 )
+
 list(APPEND CC_EXTERNAL_INCLUDES ${CMAKE_CURRENT_LIST_DIR}/include/glslang)
-set(glslang_libs_name glslang OGLCompiler OSDependent SPIRV)
+set(glslang_libs_name 
+debug glslangd optimized glslang 
+debug HLSLd optimized HLSL 
+debug OGLCompilerd optimized OGLCompiler 
+debug OSDependentd optimized OSDependent 
+debug SPIRVd optimized SPIRV 
+debug SPIRVRemapperd optimized SPIRVRemapper
+)
```

Comment:
Use this to distinguish debug and release mode?

Reply:
IMPORTED_LOCATION doesn't support generator expressions so this is what I can come up with